bin/hdfcoinc/pycbc_multiifo_add_statmap

```diff
@@ -0,0 +1,214 @@
+                          for tc in zip(*times_tuple)])
+
+is_in_combo_time = {}
+for key in f['segments']:
+    is_in_combo_time[key] = np.zeros(n_triggers)
+    if key.startswith('foreground') or key.startswith('background'):
+        continue
+    end_times = np.array(f['segments/%s/end' % key][:])
+    start_times = np.array(f['segments/%s/start' % key][:])
+    idx_within_segment = pycbc.events.indices_within_times(test_times,
+                                                           start_times,
+                                                           end_times)
+    is_in_combo_time[key][idx_within_segment] = np.ones_like(idx_within_segment)
+del idx_within_segment
+
+available_combos =[' '.join(sorted([key for key in is_in_combo_time if is_in_combo_time[key][i]])).encode('utf8') for i in np.arange(n_triggers)]
+
+all_combo_types = np.unique(available_combos)
+idx = {ct:np.where(np.array(available_combos)==ct)[0]
+       for ct in all_combo_types}
+
+del available_combos
+
+logging.info('Calculating n_louder background triggers in each type for all foreground events')
+
+far = {}
+far_exc = {}
+for f_in in files:
+        ifo_combo_key = f_in.attrs['ifos'].replace(' ','')
+        _, fnlouder = coinc.calculate_n_louder(f_in['background/stat'][:],
+                                               f['foreground/stat'][:],
+                                               f_in['background/decimation_factor'][:])
+        far[ifo_combo_key] = (fnlouder + 1) / f_in.attrs['background_time']
+        _, fnlouder_exc = coinc.calculate_n_louder(f_in['background_exc/stat'][:],
+                                                   f['foreground/stat'][:],
+                                                   f_in['background_exc/decimation_factor'][:])
+        far_exc[ifo_combo_key] = (fnlouder_exc + 1) / f_in.attrs['background_time_exc']
+
+logging.info('Recalculating ifar according to summed trigger distributions')
+
+fg_ifar = np.zeros_like(f['foreground/decimation_factor'][:])
+fg_ifar_exc = np.zeros_like(f['foreground/decimation_factor'][:])
+
+for ct in all_combo_types:
+    cts = ct.split(' ')
+    if len(cts) == 1:
+        logging.info('Copying over triggers which are only in {} time'.format(ct))
+        # If only one combination is available, then the stat is the same
+        # as previously calulated
+        fg_ifar[idx[ct]] = f['foreground/ifar'][:][idx[ct]]
+        fg_ifar_exc[idx[ct]] = f['foreground/ifar_exc'][:][idx[ct]]
+        continue
+    logging.info('Recalculating ifar for coincidences which are in {} time'.format(ct))
+    far_new = np.sum([far[ifo_combo_key][idx[ct]] for ifo_combo_key in cts], axis=0)
+    far_new_exc = np.sum([far_exc[ifo_combo_key][idx[ct]] for ifo_combo_key in cts], axis=0)
+    ifar = 1 / np.array(far_new)
+    ifar_exc = 1 / np.array(far_new_exc)
+    fg_ifar[idx[ct]] = conv.sec_to_year(ifar)
+    fg_ifar_exc[idx[ct]] = conv.sec_to_year(ifar_exc)
```

Comment:
i think these lines could be combined with the previous 2, ie `fg_ifar[idx[ct]] = conv.sec_to_year(1. / np.array(far_new))`